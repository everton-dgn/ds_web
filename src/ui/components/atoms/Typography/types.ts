export type Align = 'left' | 'center' | 'right'

export type Typography = 'headline' | 'paragraph' | 'caption'

export type TypographyProps = {
  text: string
  type: Typography
  align?: Align
}
