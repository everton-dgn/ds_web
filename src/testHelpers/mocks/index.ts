export * from './matchMedia'
export * from './reactRouterDom'
