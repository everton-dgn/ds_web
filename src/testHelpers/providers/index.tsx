export * from './providerComponents'
export * from './providerHooks'
