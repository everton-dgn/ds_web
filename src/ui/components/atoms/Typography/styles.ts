import styled, { css } from 'styled-components'
import T from 'ui/theme'
import { Align, Typography } from './types'

const typography = {
  headline: css`
    font-size: ${T.fonts.sizes.headline};
    font-weight: ${T.fonts.weight.medium};
    line-height: 29px;
  `,
  paragraph: css`
    font-size: ${T.fonts.sizes.paragraph};
    font-weight: ${T.fonts.weight.regular};
    line-height: 17px;
  `,
  caption: css`
    font-size: ${T.fonts.sizes.caption};
    font-weight: ${T.fonts.weight.regular};
    line-height: 16px;
  `
}

export const Text = styled.p<{ align: Align; type: Typography }>`
  display: flex;
  width: 100%;
  text-align: ${({ align }) => align};
  padding: 0;
  margin: 0;
  ${({ type }) => typography[type]}}
`
