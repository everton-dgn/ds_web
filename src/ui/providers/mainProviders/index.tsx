import { GlobalStyles } from 'ui/globalStyles'
import * as C from 'ui/components'
import { MainProvidersProps } from './types'

export const MainProviders = ({ children }: MainProvidersProps) => (
  <C.ErrorBoundary>
    {children}
    <GlobalStyles />
  </C.ErrorBoundary>
)
