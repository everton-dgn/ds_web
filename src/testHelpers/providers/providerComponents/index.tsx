import { render, RenderResult } from '@testing-library/react'
import { ReactNode } from 'react'
import { MainProviders } from 'ui/providers'

export const renderWithProviders = (children: ReactNode): RenderResult =>
  render(<MainProviders>{children}</MainProviders>)
