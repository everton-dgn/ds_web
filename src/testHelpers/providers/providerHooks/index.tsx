import { renderHook } from '@testing-library/react'
import { MainProviders } from 'ui/providers'
import { RenderHooksProviderProps, WrapperProps } from './types'

const wrapper = ({ children }: WrapperProps) => (
  <MainProviders>{children}</MainProviders>
)

export const renderHooksProvider = (children: RenderHooksProviderProps) =>
  renderHook(children, { wrapper })
