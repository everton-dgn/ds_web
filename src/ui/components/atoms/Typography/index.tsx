import * as S from './styles'
import { TypographyProps } from './types'

const Typography = ({ text, align = 'center', type }: TypographyProps) => {
  const as = type === 'headline' ? 'h1' : 'p'

  return (
    <S.Text as={as} align={align} type={type}>
      {text}
    </S.Text>
  )
}

export default Typography
