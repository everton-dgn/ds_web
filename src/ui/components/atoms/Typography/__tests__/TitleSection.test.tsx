import { screen } from '@testing-library/react'
import { renderWithProviders } from 'testHelpers/providers'
import Typography from '..'
import T from 'ui/theme'

describe('[Component] Typography', () => {
  it('should render a snapshot', () => {
    const { container } = renderWithProviders(
      <Typography text="Text of example" type="headline" />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render a text of type "headline"', () => {
    renderWithProviders(<Typography text="Text of example" type="headline" />)
    const headline = screen.getByRole('heading', { name: 'Text of example' })
    expect(headline).toBeInTheDocument()
    expect(headline).toHaveStyle({
      fontSize: T.fonts.sizes.headline,
      fontWeight: T.fonts.weights.medium,
      lineHeight: '29px'
    })
  })

  it('should render a text of type "paragraphHighlight"', () => {
    renderWithProviders(
      <Typography text="Text of example" type="paragraphHighlight" />
    )
    const paragraphHighlight = screen.getByText('Text of example')
    expect(paragraphHighlight).toHaveStyle({
      fontSize: T.fonts.sizes.paragraphHighlight,
      fontWeight: T.fonts.weights.medium,
      lineHeight: '17px'
    })
  })

  it('should render a text of type "paragraph"', () => {
    renderWithProviders(<Typography text="Text of example" type="paragraph" />)
    const paragraph = screen.getByText('Text of example')
    expect(paragraph).toHaveStyle({
      fontSize: T.fonts.sizes.paragraph,
      fontWeight: T.fonts.weights.regular,
      lineHeight: '17px'
    })
  })

  it('should render a text of type "caption"', () => {
    renderWithProviders(<Typography text="Text of example" type="caption" />)
    const caption = screen.getByText('Text of example')
    expect(caption).toHaveStyle({
      fontSize: T.fonts.sizes.caption,
      fontWeight: T.fonts.weights.regular,
      lineHeight: '16px'
    })
  })

  it('should render a text black center align', () => {
    renderWithProviders(<Typography text="Text of example" type="caption" />)
    const text = screen.getByText('Text of example')
    expect(text).toHaveStyle({
      color: T.colors.black,
      textAlign: 'center',
      justifyContent: 'center'
    })
  })

  it('should render a text red left aligned', () => {
    renderWithProviders(
      <Typography
        text="Text of example"
        type="caption"
        color="error"
        align="left"
      />
    )
    const text = screen.getByText('Text of example')
    expect(text).toHaveStyle({
      color: T.colors.error,
      textAlign: 'left',
      justifyContent: 'left'
    })
  })
})
