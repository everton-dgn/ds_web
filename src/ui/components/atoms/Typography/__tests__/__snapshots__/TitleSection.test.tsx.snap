// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Typography should render a snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  font-size: 24px;
  font-weight: 500;
  line-height: 29px;
  color: #282828;
}

<div>
  <h1
    class="c0"
    color="black"
    type="headline"
  >
    Text of example
  </h1>
</div>
`;
